test/rendition.content.test.js

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-env mocha */
+/* eslint mocha/no-mocha-arrows: "off" */
+
+'use strict';
+
+const assert = require('assert');
+const Rendition = require('../lib/rendition.js');
+
+const rewire = require("rewire");
+const rewiredRendition = rewire('../lib/rendition.js');
+
+// WARNING: filesystem is not mocked here, so content-type identification can be tested with known files.
+
+describe("rendition.js - content types", () => {
```

Comment:
Some content type related tests are over in `rendition.test.js` such as `handles gracefully reading incomplete data for mime+encoding`. I would have them all in one place, e.g. inside `rendition.tests.js`.

Reply:
Oh, that was on purpose. I had some issues because of the mocked file system, so the `.content`-tests do not use mockFs. I noticed that mockFs `.restore` is actually not that immediate and I had flaky tests when tests without mockFs were in the same file. It's more a workaround than a true fix for mockFs.

Reply:
> I noticed that mockFs .restore is actually not that immediate

That can't be. A missing `await` maybe?

Another solution is to make mockFs more selective on each test, not "blindly" on all.

Reply:
`make mockFs more selective on each test that requires it` <= we had issues before (not this PR) and tried this. This made it worse. It's something we will need to investigate.

Reply:
But that must have been other specific reasons of other projects and their tests.

Generally, we have been doing a lot of prepration in `beforeEach()` which applies to all tests. But this becomes problematic the more tests you add that might want a different environment - this here is one example. This apples to all kinds of setup, not just mockFs.

What I learned is better is to put all the setup in the individual tests. If the setup code is reused, put it in a shared function that those tests that need it can reuse it.